cmd/block-generator/generator/generate.go

```diff
@@ -450,33 +451,44 @@ func (g *generator) generatePaymentTxn(round uint64, intra uint64) (transactions
 	return g.generatePaymentTxnInternal(selection.(TxTypeID), round, intra)
 }
 
+const minBal = uint64(100000)
+
 func (g *generator) generatePaymentTxnInternal(selection TxTypeID, round uint64, intra uint64) (transactions.SignedTxn, transactions.ApplyData, error) {
 	defer g.recordData(track(selection))
 
+	// amounts
+	amount := uint64(1)
+	fee := uint64(1000)
+
+	// Select a receiver
 	var receiveIndex uint64
 	switch selection {
 	case paymentTx:
 		receiveIndex = rand.Uint64() % g.numAccounts
 	case paymentAcctCreateTx:
+		// give new accounts extra algos for sending other transactions (optin/optout)
+		amount = minBal * 100
 		g.balances = append(g.balances, 0)
-		g.numAccounts++
-		receiveIndex = g.numAccounts - 1
+		receiveIndex = g.numAccounts
+		// increment at the end in case it needs to be referenced later.
+		defer func() {
+			g.numAccounts++
+		}()
 	}
+	total := amount + fee
 
-	// Always send from a genesis account.
-	sendIndex := g.numPayments % g.config.NumGenesisAccounts
+	// Select a sender from genesis account
+	sendIndex := (g.numPayments + g.paymentOffset) % g.config.NumGenesisAccounts
+	// if the genesis account has insufficient balance... start checking others
+	for n := uint64(1); g.balances[sendIndex] < (total + minBal); n++ {
```

Comment:
this loop might be very slow, I'm not sure I would trust performance measurements if we got here